### DISCUSSION TITLE
RFC 1: Dates and times setting organisation

### DISCUSSION THREAD

**Discussion by orzechdev** [mirumee/saleor-dashboard#2172] [RFC / Ideas]
# Objective

There should be possibilities to set dates and/or times accordingly to each specific use case in the dashboard.

# Motivation

Currently in some places of the dashboard there are used date fields and in some date with time fields, while backend supports date with time values everywhere since https://github.com/saleor/saleor/pull/9581. It is not clearly determined where we want to provide possibility to insert date with time and where only date.

# Assumptions Proposal

Possibility to set `DATE with TIME` when:
- product is published (currently `only DATE` ❌)
- product is available (currently `only DATE` ❌)
- collection is published (currently `only DATE` ❌)
- page is published (currently `only DATE` ❌)
- active state of sale starts (currently `DATE with TIME` ✅)
- active state of sale ends (currently `DATE with TIME` ✅)
- active state of voucher starts (currently `DATE with TIME` ✅)
- active state of voucher ends (currently `DATE with TIME` ✅)

Possibility to set `only DATE` when:
- gift card expires (currently `only DATE` ✅)

# Questions

Are the assumptions correct or do we want to change some?

**Comment by dominik-zeglen** [comment]
Well I think if backend supports datetime then we should display datetime ;)

The problem I see are the timezones - should we always assume that it's UTC?
If not, then should we cast to users timezone, or servers? Or maybe we should add dropdown to select correct one?

**Reply by timuric** [reply]
yes, letting user to select timezone would be great

**Reply by orzechdev** [reply]
Ok, then where should we place that select dropdown? In user settings? Should it be saved on the backend or locally in local storage?

**Reply by timuric** [reply]
What about local storage settings that would be updated in the picker when you choose the time zone?

**Comment by supersilverlemonhaze** [comment]
Anything new here? I tried to set timezone using TZ container environment it works for everything except Saleor. How to set the timezone???
